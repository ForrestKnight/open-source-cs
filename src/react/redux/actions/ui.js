export const navMenu_Mobile__Toggle = () => ({
    type: 'MOBILE_NAV__TOGGLE'
});


export const navMenu_Mobile__On = () => ({
    type: 'MOBILE_NAV__ON'
});


export const navMenu_Mobile__Off = () => ({
    type: 'MOBILE_NAV__OFF'
});