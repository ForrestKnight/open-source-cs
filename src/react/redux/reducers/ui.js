const UIDefaultState = {
    navMenu_Mobile: false
};


export default (state = UIDefaultState, action) => {
    switch(action.type) {
        case 'MOBILE_NAV__TOGGLE':
            return {
                ...state,
                navMenu_Mobile: state.navMenu_Mobile ? false : true
            };

        case 'MOBILE_NAV__ON':
            return {
                ...state,
                navMenu_Mobile: true
            };
    

        case 'MOBILE_NAV__OFF':
            return {
                ...state,
                navMenu_Mobile: false
            };

        default:
            return state;
    };
};