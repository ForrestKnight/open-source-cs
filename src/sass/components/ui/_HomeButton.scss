#HomeButton {
    position: fixed;
    top: 10px;
    left: 10px;
    border-radius: 50%;
    z-index: 8;

    @media #{$tablet_breakpoint} {
        top: 1em;
        left: 1em;
    }
}


.HomeButton__logo {
    img {
        width: 40px;
        height: auto;
        margin: 3px 4.5px;
        animation: spin;
        -webkit-animation: spin;
        animation-iteration-count: infinite;
        animation-duration: 5s;
        animation-timing-function: linear;
    }

    @media #{$tablet_breakpoint} {
        img {
            width: 50px;
            height: auto;
            margin: 3px 4.5px;    
        }
    
        &:hover {
            img {
                animation-duration: 2s;
            }
        }
    }
}


@keyframes spin {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}

@-webkit-keyframes spin {
    0% {
        -webkit-transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
    }
}