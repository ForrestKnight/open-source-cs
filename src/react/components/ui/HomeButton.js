/* Packages */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';


/* Actions */
import {navMenu_Mobile__Toggle} from '../../redux/actions/ui';


/* Component */
export class HomeButton extends React.Component {
    constructor(props) {
        super(props);
    };


    navMenu_Mobile__Toggle = () => {
        /* Enable scrolling */
        const html = document.getElementsByTagName('HTML')[0];
        html.setAttribute('data-Mobile_Nav', 'false');
        
        this.props.navMenu_Mobile__Toggle();
    };


    render() {
        return (
            <section 
                id="HomeButton"
                onClick={this.props.navMenu_Mobile ? this.navMenu_Mobile__Toggle : undefined}
            >
                <NavLink 
                    to="/"
                    className="HomeButton__logo"
                >
                    {this.props.navMenu_Mobile ?
                        /* Nav Menu HomeButton */
                        <img src="/dist/images/homeButton/HomeButtonMobileNav.png" alt="Home"/>
                        :

                        /* Default HomeButton */
                        <img src="/dist/images/homeButton/HomeButton.png" alt="Home"/>
                    }
                </NavLink>
            </section>
        );
    };
};


/* Connect to store */
const mapStateToProps = (state) => {
    return {
        navMenu_Mobile: state.ui.navMenu_Mobile
    };
};


const mapDispatchToProps = (dispatch) => ({
    navMenu_Mobile__Toggle: () => {
        dispatch(navMenu_Mobile__Toggle());
    }
});


export default connect(mapStateToProps, mapDispatchToProps)(HomeButton);