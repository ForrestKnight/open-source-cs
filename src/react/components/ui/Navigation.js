/* Packages */
import React from 'react';


/* Components */
import HomeButton from './HomeButton';
import DesktopNavigation from './DesktopNavigation/DesktopNavigation';
import MobileNavigation from './MobileNavigation/MobileNavigation';


/* Config */
import links from '../../config/navigation';


/* Component */
export class Navigation extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <section id="Navigation">
                <HomeButton />

                <DesktopNavigation 
                    links={links}
                />
                
                <MobileNavigation 
                    links={links}                  
                />
            </section>
        );
    };
};


export default Navigation;