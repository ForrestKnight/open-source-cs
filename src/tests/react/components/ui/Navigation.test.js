/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {Navigation} from '../../../../react/components/ui/Navigation';


/* Test Config */
let component;
beforeEach(() => {
    component = shallow(<Navigation />);
});


/* Tests */
test('Render Navigation component', () => {
    expect(component).toMatchSnapshot();
});

