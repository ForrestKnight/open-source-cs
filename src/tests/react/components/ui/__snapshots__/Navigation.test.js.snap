// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navigation component 1`] = `
<section
  id="Navigation"
>
  <Connect(HomeButton) />
  <DesktopNavigation
    links={
      Array [
        Object {
          "href": "/",
          "internal": true,
          "text": "Home",
        },
        Object {
          "href": "/StyleGuide",
          "internal": true,
          "text": "Style Guide",
        },
        Object {
          "href": "https://bergers.dev",
          "internal": false,
          "text": "Meet the Dev",
        },
        Object {
          "href": "https://github.com/RobbyB97/JavascriptReactBoilerplate",
          "internal": false,
          "text": "View on GitHub",
        },
      ]
    }
  />
  <Connect(MobileNavigation)
    links={
      Array [
        Object {
          "href": "/",
          "internal": true,
          "text": "Home",
        },
        Object {
          "href": "/StyleGuide",
          "internal": true,
          "text": "Style Guide",
        },
        Object {
          "href": "https://bergers.dev",
          "internal": false,
          "text": "Meet the Dev",
        },
        Object {
          "href": "https://github.com/RobbyB97/JavascriptReactBoilerplate",
          "internal": false,
          "text": "View on GitHub",
        },
      ]
    }
  />
</section>
`;
